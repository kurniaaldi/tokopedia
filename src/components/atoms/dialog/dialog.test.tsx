import { render, screen } from "@testing-library/react";
import Dialog from "./index";
import renderer from "react-test-renderer";

test("dialog", () => {
  const Component = () => <p>card</p>;

  let data = {
    open: true,
    handleClose: () => (data = { ...data, open: false }),
    title: "header",
  };

  render(
    <Dialog title={data.title} handleClose={data.handleClose} open={data.open}>
      <Component />
    </Dialog>
  );

  const dialogElement = screen.getByTestId("dialog-id");
  expect(dialogElement).toBeInTheDocument();
  expect(dialogElement).toHaveTextContent("header");
  expect(dialogElement).toContainHTML(
    `<div style="width: 100%; display: flex; align-items: center; justify-content: center; padding: 1rem 1rem 0px 1rem;">header</div><main class="modal__main"><p>card</p></main>`
  );
});

test("matches snapshot", () => {
  const Component = () => <p>card</p>;

  let data = {
    open: true,
    handleClose: () => (data = { ...data, open: false }),
    title: "header",
  };

  const tree = renderer
    .create(
      <Dialog
        title={data.title}
        handleClose={data.handleClose}
        open={data.open}
      >
        <Component />
      </Dialog>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
