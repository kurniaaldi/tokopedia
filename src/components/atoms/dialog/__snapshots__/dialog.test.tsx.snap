// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="overlay"
    onClick={[Function]}
  />,
  <div
    className="modal"
    data-testid="dialog-id"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "padding": "1rem",
          "paddingBottom": "0",
          "width": "100%",
        }
      }
    >
      header
    </div>
    <main
      className="modal__main"
    >
      <p>
        card
      </p>
    </main>
  </div>,
]
`;
